tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestUnits.java

```diff
@@ -4245,4 +4270,337 @@ public <T> boolean assertNotContains(
+            Files.lines(Path.of(CLDRPaths.TEST_DATA + "units/unitPreferencesTest.txt"))
+                    .forEach(line -> checkUnitPreferencesTest(line, warnings));
+            if (!warnings.isEmpty()) {
+                warnln("Mixed unit identifiers not yet checked, count=" + warnings.size());
+            }
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    public void checkUnitPreferencesTest(String line, Set<String> warnings) {
+        if (line.startsWith("#") || line.isBlank()) {
+            return;
+        }
+        // #    Quantity;   Usage;  Region; Input (r);  Input (d);  Input Unit; Output (r);
+        // Output (d); Output Unit
+        // Example:
+        // area;      default;    001;    1100000;    1100000.0;  square-meter;
+        // 11/10;  1.1;    square-kilometer
+        // duration;   media;     001;    66;         66.0;       second;        1; minute;   6;
+        //      6.0;    second
+        try {
+            UnitPreferences prefs = SDI.getUnitPreferences();
+            List<String> parts = SPLIT_SEMI.splitToList(line);
+            Map<String, Long> highMixed_unit_identifiers = new LinkedHashMap<>();
+            String quantity = parts.get(0);
+            String usage = parts.get(1);
+            String region = parts.get(2);
+            Rational inputRational = Rational.of(parts.get(3));
+            double inputDouble = Double.parseDouble(parts.get(4));
+            String inputUnit = parts.get(5);
+            // account for multi-part output
+            int size = parts.size();
+            // This section has larger elements with integer values
+            for (int i = 6; i < size - 3; i += 2) {
+                highMixed_unit_identifiers.put(parts.get(i + 1), Long.parseLong(parts.get(i)));
+            }
+            Rational expectedValue = Rational.of(parts.get(size - 3));
+            Double expectedValueDouble = Double.parseDouble(parts.get(size - 2));
+            String expectedOutputUnit = parts.get(size - 1);
+
+            // Check that the double values are approximately the same as
+            // the Rational ones
+            assertTrue(
+                    String.format(
+                            "input rational ~ input double, %s %s", inputRational, inputDouble),
+                    inputRational.approximatelyEquals(inputDouble));
+            assertTrue(
+                    String.format(
+                            "output rational ~ output double, %s %s",
+                            expectedValue, expectedValueDouble),
+                    expectedValue.approximatelyEquals(expectedValueDouble));
+
+            // check that the quantity is consistent
+            String expectedQuantity = converter.getQuantityFromUnit(inputUnit, false);
+            assertEquals("Input: Quantity consistency check", expectedQuantity, quantity);
+
+            // TODO handle mixed_unit_identifiers
```

Comment:
TODO should have ticket number

Reply:
I'd like to do that in a subsequent PR. I had to squash, and it is much simpler to do in a follow-on PR.